### PR TITLE
First Implementation of Google Dagger for Android

### PR DESCRIPTION
**Pull request by Iguins**
Google Dagger is required for Firebase InAppMessaging in Xamarin GooglePlayServiceComponent Rep.

### CODE REVIEW

File: Android/GoogleDagger/source/Dagger/Transforms/Metadata.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<metadata>
+	<attr path="/api/package[@name='com.google.dagger.dagger']" name="managedName">Google.Dagger.Dagger</attr>
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Transforms/Metadata.xml]: I think maybe we should be ok with just `Google.Dagger` as the namespace.

File: Android/GoogleDagger/source/Dagger/Dagger.csproj [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Dagger.csproj]: Could you make this `<Project Sdk="MSBuild.Sdk.Extras/2.0.54">` to use the newer sdk extras package?  Also see some comments below to remove unnecessary bits after this change...

File: Android/GoogleDagger/source/Dagger/Dagger.csproj [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <Link>dagger.jar</Link>
+    </None>
+    <None Include="..\..\External-Dependency-Info.txt" Pack="True" PackagePath="THIRD-PARTY-NOTICES.txt" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedJar Include="..\..\externals\dagger.jar" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Dagger.csproj:47]: Remove this bit if we are using the Sdk specifier in the <Project element

File: Android/GoogleDagger/source/Dagger/Dagger.csproj [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <PackageReference Include="MSBuild.Sdk.Extras" Version="1.4.0" PrivateAssets="All" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="Mono.Android" />
+  </ItemGroup>
+  <Import Project="$(MSBuildSDKExtrasTargets)" Condition="Exists('$(MSBuildSDKExtrasTargets)')" />
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Dagger.csproj]: Remove this bit as well if we are using Sdk specifier in the <Project element...

File: Android/GoogleDagger/source/Dagger/Additions/AboutAdditions.txt [hunk 5]
```diff
@@ -0,0 +1,48 @@
+﻿Additions allow you to add arbitrary C# to the generated classes
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Additions/AboutAdditions.txt]: Could you remove this file?

File: Android/GoogleDagger/source/Dagger/Additions/Additions.cs [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿using System;
+
+namespace Google.Dagger.Dagger {
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Additions/Additions.cs]: Namespace change here, though in practice we could just remove this file altogether if it's not specifying anything

File: Android/GoogleDagger/External-Dependency-Info.txt [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+Xamarin Components for Google Dagger incorporates 
+third party material from the projects listed below. The original copyright 
+notice and the license under which Microsoft received such third party 
+material are set forth below.  Microsoft reserves all other rights not 
+expressly granted, whether by implication, estoppel or otherwise.
+
+########################################
+# Google Dagger
+# https://github.com/google/auto
```
Comment (Redth) [inline Android/GoogleDagger/External-Dependency-Info.txt]: URL I think should be: https://github.com/google/dagger

**Review by Redth — CHANGES_REQUESTED** [review]
Thanks so much for this!  A few minor changes requested, but awesome work on it, and much appreciated!

**Comment by moljac** [comment]
Seems like nuget download of nuget validation cake script is failing

```
Downloading script from https://raw.githubusercontent.com/xamarin/XamarinComponents/GoogleDaggerAndroid/.ci/validation.cake...
Invoke-WebRequest : 404: Not Found
```

File: Android/GoogleDagger/source/Dagger/Dagger.csproj [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PackageId>Xamarin.Google.Dagger.Dagger</PackageId>
+    <Title>Google Dagger for Xamarin.Android</Title>
+    <Summary>Xamarin.Android bindings for Google Dagger</Summary>
+    <Authors>Microsoft</Authors>
+    <Owners>Microsoft</Owners>
+    <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
+    <PackageProjectUrl>https://go.microsoft.com/fwlink/?linkid=2091414</PackageProjectUrl>
```
Comment (Redth) [inline Android/GoogleDagger/source/Dagger/Dagger.csproj]: Can you leave these nodes both empty for now?  These links aren't correct but we will fill them out properly after :)

**Comment by Redth** [comment]
Thank you!!

**Pull request merged**